src/adapters/fastify.ts

```diff
@@ -37,7 +37,7 @@ export function handleFastifyReply(res: FastifyReply, response: OAuthResponse):
   if (response.status === 302) {
     if (!response.headers.location) throw new Error("missing redirect location");
     res.headers(response.headers);
-    res.redirect(response.headers.location);
+    res.redirect(response.headers.location, response.status);
```

Comment:
Since this check is only going to happen when the response status is 302, I think it might be more clear if we just hard code 302 rather than using `response.status`. What are your thoughts?